gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/events/EventWorkunitUtils.java

```diff
@@ -26,26 +29,65 @@
   /**
    * Set SLA event metadata in the workunit. The publisher will use this metadta to publish sla events
    */
-  public static void setTableSlaEventMetadata(WorkUnit state, Table table, long updateTime, long lowWatermark) {
+  public static void setTableSlaEventMetadata(WorkUnit state, Table table, long updateTime, long lowWatermark,
+      long beginGetWorkunitsTime) {
     state.setProp(SlaEventKeys.DATASET_URN_KEY, state.getProp(ConfigurationKeys.DATASET_URN_KEY));
     state.setProp(SlaEventKeys.PARTITION_KEY, table.getCompleteName());
     state.setProp(SlaEventKeys.UPSTREAM_TS_IN_MILLI_SECS_KEY, String.valueOf(updateTime));
 
     // Time when the workunit was created
     state.setProp(SlaEventKeys.ORIGIN_TS_IN_MILLI_SECS_KEY, System.currentTimeMillis());
+    state.setProp(EventConstants.WORK_UNIT_CREATE_TIME, state.getProp(SlaEventKeys.ORIGIN_TS_IN_MILLI_SECS_KEY));
```

Comment:
```BEGIN_GET_WORKUNITS_TIME``` is set when ```source.getWorkunits(SourceState state)``` is called. So it will be same for all workunits. ```WORK_UNIT_CREATE_TIME``` is the time when a workunit was created. There can be a delay between ```BEGIN_GET_WORKUNITS_TIME``` and ```WORK_UNIT_CREATE_TIME``` because we make a bunch of hive metastore calls and HDFS calls.